### PR TITLE
Add scale indexing

### PR DESCRIPTION
**Pull request by dpsanders**

### CODE REVIEW

File: src/scales.jl [hunk 1]
```diff
@@ -42,6 +46,19 @@ Base.IteratorEltype(::Type{Scale{T}}) where {T} = Base.HasEltype()
...
+
+    octave = n ÷ length(s.notes)
+    n = n % length(s.notes)
+    if n < 0
+        n += length(s.notes)
+        octave -= 1
+    end
+
+    return Pitch(PitchClass(s.notes[n + 1]), s.notes[n + 1].octave + octave)
+end
```
Comment (stevengj) [inline src/scales.jl:59]: Why zero-based indexing?

Comment (dpsanders) [reply src/scales.jl:59]: What else would make sense? That way positive and negative are symmetric around the tonic.

Comment (stevengj) [reply src/scales.jl:59]: 1-based indexing is not only more common in Julia, but it also seems more common. If you poll 100 musicians with “what is the 5th note in the C major scale”, how many do you think would answer “A”?

Comment (stevengj) [reply src/scales.jl:59]: See e.g. https://en.m.wikipedia.org/wiki/Degree_(music)

Comment (stevengj) [reply src/scales.jl:59]: Nor do understand the point of positive and negative indices being symmetric around the tonic. The intervals aren’t symmetric, nor are the musical functions of the notes.

Comment (dpsanders) [reply src/scales.jl:59]: Good points, thanks! Let me rethink

**Pull request merged**